src/kloewe/dot/src/dot.h

```diff
@@ -0,0 +1,92 @@
+/*----------------------------------------------------------------------------
+  File    : dot.h
+  Contents: dot product (cpu dispatcher)
+  Author  : Kristian Loewe
+----------------------------------------------------------------------------*/
+#ifndef DOT_H
+#define DOT_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/*----------------------------------------------------------------------------
+  Enum to encode the sets of dot implementations
+----------------------------------------------------------------------------*/
+enum flags {
+    DOT_NAIVE  = 1,
+    DOT_SSE2   = 2,
+    DOT_AVX    = 3,
+    #ifndef DOT_NOFMA
+    DOT_AVXFMA = 4,
+    #endif
+    DOT_AUTO   = 3
```

Comment:
That works, might be better than my idea.

Out of curiousity, I did a little searching on avxfma, and ran across this:

http://stackoverflow.com/questions/35859449/why-are-some-haswell-avx-latencies-advertised-by-intel-as-3x-slower-than-sandy-b

Which, in the top answer, has this relevant-sounding bit (because naively, the summation variable is a loop-carried dependency):

"Latency is more often a factor in FP code [...] [on Skylake they] managed to drop the latency for FMA to 4 cycles [...] So using many vector accumulators is essential in most FP algorithms for keeping 8 or 10 FMAs in flight at once to saturate the throughput, if there's a loop-carried dependency."